src/sensor/ping.cpp

```diff
@@ -564,6 +581,32 @@ void Ping::writeMessage(const PingMessage &msg)
     }
 }
 
+void Ping::checkNewFirmwareInGitHubPayload(const QJsonDocument& jsonDocument)
+{
+    struct {
+        float version = 0.0;
+        QString downloadUrl;
+    } versionAvailable;
+
+    auto filesPayload = jsonDocument.array();
+    for(const QJsonValue& filePayload : filesPayload) {
+        qCDebug(PING_PROTOCOL_PING) << filePayload["name"].toString();
+
+        // Get version from Ping_V(major).(patch)_115kb.hex where (major).(patch) is <version>
+        static const QRegularExpression versionRegex(QStringLiteral(R"(Ping_V(?<version>\d+\.\d+)_115kb\.hex)"));
+        versionAvailable.version = versionRegex.match(filePayload["name"].toString()).captured("version").toFloat();
```

Comment:
There is no guarantee that the json will be ordered in ascending version. You are making that assumption here, so you could just take array.last() instead of looping through each. (but let's stop making that assumption):
```suggestion
        versionAvailable = versionRegex.match(filePayload["name"].toString()).captured("version").toFloat();
        if (versionAvaliable > latestVersionAvailable.version) {
            latestVersionAvailable.version = versionAvailable;
            latestVersionAvailable.downloadUrl = ...
        }
```

Reply:
True, probably I lost some code after moving things around, this was done in the initial version. Going to fix it soon.